src/lib.rs

```diff
@@ -35,6 +35,32 @@
 
 use core::mem::size_of;
 
+/// Error types for conversions
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Error {
+    /// Source value lies outside of target type's range
+    Range,
+    /// Loss of precision and/or outside of target type's range
+    Inexact,
+}
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "easy-cast conversion error: {}",
```

Comment:
Would prefer just "cast conversion error" here, probably. Not a big deal either way.